docs/broker/README.md

```diff
@@ -51,6 +51,28 @@ spec:
 
 ## Usage
 
+### ClusterChannelProvisioner
+
+`Broker`'s use their `spec.channelTemplate` to create their internal `Channel`s, which dictate the durability guarantees of events sent to that `Broker`. If `spec.channemTemplate` is not specified, then the [default provisioner](https://github.com/knative/docs/blob/master/docs/eventing/channels/default-channels.md#setting-the-default-channel-configuration) for their namespace is used.
+
+#### Setup
+
+Have a `ClusterChannelProvisioner` installed and set as the [default provisioner](https://github.com/knative/docs/blob/master/docs/eventing/channels/default-channels.md#setting-the-default-channel-configuration) for the namespace you are interested in. For development, the [`in-memory` `ClusterChannelProvisioner`](https://github.com/knative/eventing/tree/master/config/provisioners/in-memory-channel#deployment-steps) is normally used.
+
+#### Changing
+
+**Note** changing the `ClusterChannelProvisioner` of a running `Broker` will lose all in-flight events.
+
+If you want to change which `ClusterChannelProvisioner` is used by a given `Broker`, then determine if the `spec.channelTemplate` is specified or not.
+
+If `spec.channelTemplate` is specified:
+1. Delete the `Broker`.
+1. Create the `Broker` with the updated `spec.channelTemplate`.
+
+If `spec.channelTemplate` is not specified:
```

Comment:
Is this specifically for the `default` Broker, or is is this intended as a general solution?

Reply:
General solution. Incidentally, it is the only one that will work with the `default` Broker (if auto-injected anyway), because the injected `default` does not specify `spec.channelTemplate`.

Reply:
Isn't this a bit all very manual? and might be error prone ?

Reply:
It is manual. What errors would it be prone to?

When implementing Broker, I made the decision not to update Channels even if the Broker's `spec.channelTemplate` changed because it would drop all in flight events. We can revisit that decision and have it updated on the fly. The code change would be quite small and there is already a comment in the place it would occur (see https://github.com/knative/eventing/pull/996#discussion_r271022051).

If someone _really_ wants to cause an update on the fly, they can delete the Channels backing the Broker manually. The Broker controller will create new ones with the updated template. But I don't want to document this as I consider it an implementation detail that may change in the future.